src/etc/inc/crypt.inc

```diff
@@ -39,10 +39,11 @@
 		 *  unless we need to read old data encrypted without it. */
 		$keyder = ($legacy) ? "" : "-pbkdf2";
 		$md = ($legacy) ? "md5" : "sha256";
+		$iter = ($legacy) ? "" : "-iter 300000";
```

Comment:
We should make this at least 310000 if I'm reading this right:

https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html

That's the closest I could find to a current recommendation for PBKDF2 iterations.

Would be good to add a note and link to that URL above that line so we have a reference for it to show it wasn't pulled out of thin air. If not that URL, link to the Redmine issue #.

Reply:
I updated the commit
Please note that this would make old encrypted backups non recoverable from the GUI !
I am not really sure if a GUI fix should be added for it, but would be out of my scope and time
However, clearly staying with 10000 is a bad design for a firewall

Reply:
If it would cause a failure like that, then we couldn't accept it without the extra accommodation included. There is already an example of it falling back to legacy, but basically it could do this:

* Try to decrypt with the large number of iterations.
* If that fails, try to decrypt with the default number of iterations
* If that fails, use the legacy options

What I would do is add a new parameter to `crypt_data()` which specifies the number of iterations, and then it would be easy to adapt the code to handle the extra choice.

If you aren't able to take that part on I can look into doing it next week.

Reply:
Sure please, as I am not used to pfsense sources, so it would take me too much time running through it
Hope you can follow up and do all the commits you think are needed
Best regards